pkg/controller/operands/kubevirt_test.go

```diff
@@ -499,6 +708,147 @@ var _ = Describe("KubeVirt Operand", func() {
 			Expect(req.Conditions).To(BeEmpty())
 		})
 
+		Context("Feature Gates", func() {
+			origKvFeatureGates := managedKvFeatureGates
+
+			BeforeEach(func() {
+				// list of fake KV managed FGs
+				managedKvFeatureGates = []string{fgEnabled, fgMissing, fgDisabled, fgNoChange}
+			})
+
+			AfterEach(func() {
+				managedKvFeatureGates = origKvFeatureGates
+			})
+
+			It("should add the feature gates if they are set in HyperConverged CR", func() {
+				hco.Spec.FeatureGates = map[string]bool{
+					fgEnabled:  true,
+					fgDisabled: false,
+				}
+
+				existingResource := NewKubeVirt(hco)
+				By("KV CR should contain the fgEnabled feature gate", func() {
+					Expect(existingResource.Spec.Configuration.DeveloperConfiguration).NotTo(BeNil())
+					Expect(existingResource.Spec.Configuration.DeveloperConfiguration.FeatureGates).To(ContainElement(fgEnabled))
+					Expect(existingResource.Spec.Configuration.DeveloperConfiguration.FeatureGates).ToNot(ContainElement(fgMissing))
+					Expect(existingResource.Spec.Configuration.DeveloperConfiguration.FeatureGates).ToNot(ContainElement(fgDisabled))
+				})
+			})
+
+			It("should not add the feature gates if FeatureGates map is nil", func() {
+				existingResource := NewKubeVirt(hco)
+				Expect(existingResource.Spec.Configuration.DeveloperConfiguration).To(BeNil())
+			})
+
+			It("should add feature gates if they are set to true", func() {
+				existingResource := NewKubeVirt(hco)
+
+				hco.Spec.FeatureGates = map[string]bool{
+					fgEnabled:  true,
+					fgDisabled: false,
+				}
+
+				cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+				handler := (*genericOperand)(newKubevirtHandler(cl, commonTestUtils.GetScheme()))
+				res := handler.ensure(req)
+				Expect(res.UpgradeDone).To(BeFalse())
+				Expect(res.Updated).To(BeTrue())
+				Expect(res.Overwritten).To(BeFalse())
+				Expect(res.Err).To(BeNil())
+
+				foundResource := &kubevirtv1.KubeVirt{}
+				Expect(
+					cl.Get(context.TODO(),
+						types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+						foundResource),
+				).To(BeNil())
+
+				By("KV CR should contain the WithHostPassthroughCPU feature gate", func() {
```

Comment:
I think we are missing a dedicated `Expect` to locate the `WithHostPassthroughCPU`